modules/_node-scripts/report/source/imports.mjs

```diff
@@ -0,0 +1,160 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import * as acorn from 'acorn';
+import tsPlugin from 'acorn-typescript';
+import estraverse from 'estraverse';
+import fg from 'fast-glob';
+import fs from 'fs/promises';
+import path from 'path';
+
+import {SRC_PATH, getRootDir} from '../../util/constants.mjs';
+import getYarnWorkspaceProjects from '../../util/getYarnWorkspaceProjects.mjs';
+import launchSoffice from '../../util/launchSoffice.mjs';
+
+export default async function main() {
+	const [projectDirectories, rootDir] = await Promise.all([
+		getYarnWorkspaceProjects(),
+		getRootDir(),
+	]);
+
+	const results = await Promise.all(
+		projectDirectories.map(async (projectDir) => {
+			const files = await fg(
+				`${path.join(projectDir, SRC_PATH)}/**/*.{js,jsx,ts,tsx}`
```

Comment:
mjs? or maybe we don't need that

Reply:
AFAIK we don't use `.mjs` in DXP production code, right?

Reply:
In node-scripts we do, but I guess that doesn't impact this.

Reply:
Ah OK, we do but only for implementing node-scripts and this tool doesn't analyze server side code, just the production code we run at the browser :+1: